File: calendar-backend/src/main/java/com/example/calendarbackend/model/Event.java
package com.example.calendarbackend.model;

import lombok.Getter;

import java.time.OffsetDateTime;

@Getter
public class Event {
    private Integer id;
    private String title;
    private String description;
    private OffsetDateTime startAt;
    private OffsetDateTime endAt;
    private String address;
    private int createdBy;
    private String scopeCode;
    private Integer groupId;
    private OffsetDateTime createdAt;

    public Event(String title, String description, OffsetDateTime startAt, OffsetDateTime endAt, String address, int createdBy, String scopeCode, Integer groupId)
    {
        this.title = title;
        this.description = description;
        this.startAt = startAt;
        this.endAt = endAt;
        this.address = address;
        this.createdBy = createdBy;
        this.scopeCode = scopeCode;
        this.groupId = groupId;
    }

}

File: calendar-client/lib/repositories/events/models/event.dart
import 'package:equatable/equatable.dart';

class Event extends Equatable
{
  const Event({
    required this.id,
    required this.title,
    required this.description,
    required this.startAt,
    required this.endAt,
    required this.address,
    required this.createdBy,
    required this.scopeCode,
    required this.groupId,
  });

  final int id;
  final String? title;
  final String? description;
  final DateTime startAt;
  final DateTime endAt;
  final String? address;
  final int createdBy;
  final String scopeCode;
  final int? groupId;

  @override
  List<Object?> get props => [
    id,
    title,
    description,
    startAt,
    endAt,
    address,
    createdBy,
    scopeCode,
    groupId,
  ];

  factory Event.fromJson (Map<String, dynamic> json)
  {
    return Event(
      id: json['id'] as int,
      title: json['title'] as String?,
      description: json['description'] as String?,
      startAt: DateTime.parse(json['startAt'] as String),
      endAt: DateTime.parse(json['endAt'] as String),
      address: json['address'] as String?,
      createdBy: json['createdBy'] as int,
      scopeCode: json['scopeCode'] as String,
      groupId: json['groupId'] as int?,
    );
  }
}